src/services/fastifyHooksSetUp.service.ts

```diff
@@ -0,0 +1,42 @@
+import { FastifyInstance, preSerializationHookHandler } from 'fastify';
+import { v4 as uuidv4 } from 'uuid';
+import { logger } from '../common/services';
+import { processManagerService } from './';
+
+class FastifyHookService {
+  private fastify: FastifyInstance;
+
+
+  constructor(fastify: FastifyInstance) {
+    this.fastify = fastify;
+    this.setupHooks()
+  }
+
+  public setupHooks() {
+    this.fastify.addHook("onRequest", async (request: any, reply) => {
```

Comment:
instead of nesting async functions we can simply use this
this.fastify.addHook("onRequest",  processManagerService.onRequestHander(request))
and use setimmediate in onRequestHandler function

Reply:
We'll have to pass handler function like this

this.fastify.addHook("onResponse", async (request: any, reply: any) => {
  await processManagerService.onResponseHandler(request, reply, reply.payload);
});

we cant pass it directly in second arg, it will executed immediately